src/libraries/System.Linq/tests/AllTests.cs

```diff
@@ -57,6 +57,18 @@ public void All(IEnumerable<int> source, Func<int, bool> predicate, bool expecte
             Assert.Equal(expected, source.All(predicate));
         }
 
+        [Theory]
+        [MemberData(nameof(All_TestData))]
+        public void AllList(IEnumerable<int> source, Func<int, bool> predicate, bool expected) {
+            Assert.Equal(expected, System.Linq.Enumerable.ToList(source).All(predicate));
+        }
+
+        [Theory]
+        [MemberData(nameof(All_TestData))]
+        public void AllArray(IEnumerable<int> source, Func<int, bool> predicate, bool expected) {
+            Assert.Equal(expected, System.Linq.Enumerable.ToArray(source).All(predicate));
+        }
+
```

Comment:
`All_TestData` already has arrays in it.  You should probably just change the test data to _also_ return `List`, and remove these test specializations.